### PR TITLE
Mail2kvanitha/c01-git03

### PR DESCRIPTION
**Pull request by mail2kvanitha**
# Description

Exercise (e.g. C01-E01): c01-git03

Student Name: Vanitha Kaliyaperumal

> A brief explanation of what has been done.
How to delete a large file from local and also from master repository.

## Type of change

- [ ] Exercise Submission (one exercise per PR)
  - [ ] This PR is just for **one** exercise of a class
  - [ ] PR name follows the pattern `<github-username>/<exercise-number>`
  - [ ] Added exercise files inside folder `/classes/<class- number>/exercises/<exercise-number>/<github-username>/`
	- [ ] Review [exercise submission steps](./README.md#Exercises)
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/01class/exercises/c01-git01/mail2kvanitha/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# c01-git01
```
Comment (kikobr82) [inline classes/01class/exercises/c01-git01/mail2kvanitha/README.md:1]: This file should not be part of the c01-git03 exercise delivery

File: classes/01class/exercises/c01-git02/mail2kvanitha/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# c01-git02
```
Comment (kikobr82) [inline classes/01class/exercises/c01-git02/mail2kvanitha/README.md:1]: This file should not be part of the c01-git03 exercise delivery

**Comment by kikobr82** [comment]
what's the error message when trying to remove that file?

File: classes/01class/exercises/c01-git01/mail2kvanitha/README.md (hunk 1 above)
Comment (mail2kvanitha) [reply classes/01class/exercises/c01-git01/mail2kvanitha/README.md]: Removed c01-git01 file from c01-giit03 branch

File: classes/01class/exercises/c01-git02/mail2kvanitha/README.md (hunk 2 above)
Comment (mail2kvanitha) [reply classes/01class/exercises/c01-git02/mail2kvanitha/README.md]: removed c01-git02 file from c01-git03 branch

File: classes/01class/exercises/c01-git01/mail2kvanitha/README.md (hunk 1 above)
Comment (mail2kvanitha) [reply classes/01class/exercises/c01-git01/mail2kvanitha/README.md]: projects/project02/application/web/public/favicon.ico file removed, Please advise if there is anything pending.

File: classes/01class/exercises/c01-git03/mail2kvanitha/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+## Questions (answer inline)
+
+1. Did you notice any difference in the size of the repo before and after adding the big file?
+The size of the repo increased significantly after adding the big file
+
+2. What is the reason for this problem?
+Git maintains the history of commits of each file/branches, and hence the repo size grows with each commit.
+
+3. How do you remove something from Git history after it is pushed to the remote repository? Which commands would you use? 
+"git filter-branch --index-filter "git rm -rf --cached --ignore-unmatch path_to_file HEAD" to remove file from local repository
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/mail2kvanitha/README.md:12]: That works too, but the new recommended tool for this is `git filter-repo`. It is simpler to use and safer.

File: classes/01class/exercises/c01-git03/mail2kvanitha/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+"git push origin --force --all" to remove file from Remote Origin Master
+
+4. What are the consequences of the previous action?
+The --index-filter in the above command will run against the index at the each step in the history. The "rm -rf --cached --ignore-unmatch" will remove the matching filename forcefully. 
+The second command will remove the files from remote origin.
```
Comment (denstorti) [inline classes/01class/exercises/c01-git03/mail2kvanitha/README.md:17]: That's what the commands do, right.

But the consequences for other developers that created branches out of this one is that this command will rewrite the commit history and they will need to sync their dev branches (with rebase or merge)

**Pull request merged**